content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-sample.md

```diff
@@ -0,0 +1,641 @@
+
+The three streaming applications are as follows:
+
+- The `Source` application (named `UsageDetailSender`) generates the users' call `duration` and the amount of `data` used for each `userId` and sends a message that contains the `UsageDetail` object as JSON.
+
+- The `Processor` application (named `UsageCostProcessor`) consumes the `UsageDetail` and computes the cost of the call and the cost of the data per `userId`. It sends the `UsageCostDetail` object as JSON.
+
+- The `Sink` application (named `UsageCostLogger`) consumes the `UsageCostDetail` object and logs the cost of the call and data.
+
+## Source
+
+In this step, we create the `UsageDetailSender` source.
+
+<!--TABS-->
+
+<!--Kafka-->
+
+You can either [directly download the project generated by the Spring Initialzr](https://start.spring.io/#!type=maven-project&language=java&platformVersion=2.4.3.RELEASE&packaging=jar&jvmVersion=1.8&groupId=io.spring.dataflow.sample&artifactId=usage-detail-sender-kafka&name=usage-detail-sender-kafka&description=Demo%20project%20for%20Spring%20Boot&packageName=io.spring.dataflow.sample.usagedetailsender&dependencies=cloud-stream,actuator,kafka)
+
+or visit the [Spring Initialzr site](https://start.spring.io/) and follow these instructions:
+
+1. Create a new Maven project with a Group name of `io.spring.dataflow.sample` and an Artifact name of `usage-detail-sender-kafka`, and package `o.spring.dataflow.sample.usagedetailsender`.
+2. In the **Dependencies** text box, type `Kafka` to select the Kafka binder dependency,.
+3. In the **Dependencies** text box, type `Cloud Stream` to select the Spring Cloud Stream dependency.
+4. In the **Dependencies** text box, type `Actuator` to select the Spring Boot actuator dependency.
+5. Click the **Generate Project** button.
+
+Now you should `unzip` the `usage-detail-sender-kafka.zip` file and import the project into your favorite IDE.
+
+There are many configuration options that you can choose to extend or override to achieve the desired runtime behavior when using Kafka as the message broker. The [Kafka-binder documentation](There are many configuration options that you can choose to extend or override to achieve the desired runtime behavior when using RabbitMQ as the message broker. The [RabbitMQ-binder documentation](https://docs.spring.io/spring-cloud-stream-binder-kafka/docs/3.1.1/reference/html/spring-cloud-stream-binder-kafka.html#_configuration_options) lists the Kafka binder configuration properties.
+
+<!--RabbitMQ-->
+
+You can either [directly download the project generated by the Spring Initialzr](https://start.spring.io/#!type=maven-project&language=java&platformVersion=2.4.3.RELEASE&packaging=jar&jvmVersion=1.8&groupId=io.spring.dataflow.sample&artifactId=usage-detail-sender-rabbit&name=usage-detail-sender-rabbit&description=Demo%20project%20for%20Spring%20Boot&packageName=io.spring.dataflow.sample.usagedetailsender&dependencies=cloud-stream,actuator,amqp)
```

Comment:
Takes you to start.spring.io with the site properly populated, but you have to click the generate button to download the page.